agent/worker.go

```diff
@@ -568,9 +573,9 @@ func (w *worker) Subscribe(ctx context.Context, subscription *api.SubscriptionMe
 		case v := <-ch:
 			task := v.(*api.Task)
 			if match(task) {
-				w.mu.Lock()
+				w.mu.RLock()
```

Comment:
It happens before the goroutine launches.

https://golang.org/ref/spec#Go_statements